js/index.js

```diff
@@ -4,20 +4,37 @@ function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
 
   //... your code goes here
+  const price = product.querySelector('.price span').innerText;
+  const quantity = product.querySelector('.quantity input').value;
+
+  let subtotal = price * quantity;
+  product.querySelector('.subtotal > span').innerText = subtotal;
+  return subtotal;
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+  // const singleProduct = document.querySelector('.product');
+  // updateSubtotal(singleProduct);
   // end of test
 
   // ITERATION 2
   //... your code goes here
 
+  // I need to itinerate throughout the table and get the name of the product, the price and the quantity
+    for ( product of document.getElementsByClassName('product')) {
```

Comment:
You don't need this loop, because the second loop also updates the subtotal and calculates the total price.